src/render/shaders/Textures.hpp

```diff
@@ -3,138 +3,38 @@
+	vec2 pixCoord = v_texcoord - vec2(0.5);
+	pixCoord *= (vec2(lessThan(pixCoord, vec2(0.0))) * vec2(-2.0) + vec2(1.0)) * fullSize;
+	pixCoord -= (bottomRight - topLeft) * vec2(0.5);
 
-			if (ignoreCorners == 1) {
-				discard;
-				return;
-			}
-
-			float topLeftDistance = distance(vec2(bottomRight[0], topLeft[1]), pixCoord);
-
-			if (topLeftDistance > radius - 1.0) {
-				if (primitiveMultisample == 0 && topLeftDistance > radius) {
-					discard;
-					return;
-				} else if (primitiveMultisample == 1) {
-					float distances = 0.0;
-					if (distance(vec2(bottomRight[0], topLeft[1]), pixCoord + vec2(0.25, 0.25)) < radius) { distances = distances + 1.0; }
-					if (distance(vec2(bottomRight[0], topLeft[1]), pixCoord + vec2(0.75, 0.25)) < radius) { distances = distances + 1.0; }
-					if (distance(vec2(bottomRight[0], topLeft[1]), pixCoord + vec2(0.25, 0.75)) < radius) { distances = distances + 1.0; }
-					if (distance(vec2(bottomRight[0], topLeft[1]), pixCoord + vec2(0.75, 0.75)) < radius) { distances = distances + 1.0; }
-
-					if (distances == 0.0) {
-						discard;
-						return;
-					}
-
-					distances = distances / 4.0;
-
-					)#" + colorVarName + R"#( = )#" + colorVarName + R"#( * distances;
-				}
-			}
-		} else if (pixCoord[1] > bottomRight[1]) {
-			// bottom
-
-			if (ignoreCorners == 1) {
+	if (all(greaterThan(pixCoord, vec2(0.0)))) {
+
+		if (ignoreCorners == 1)
+			discard;
+
+		float dist = length(pixCoord);
+
+		if (dist > radius)
+			discard;
+
+		if (primitiveMultisample == 1 && dist > radius - 1.0) {
+			float distances = 0.0;
+			if (length(pixCoord - vec2(0.25, 0.25)) < radius) { distances = distances + 1.0; }
+			if (length(pixCoord - vec2(0.75, 0.25)) < radius) { distances = distances + 1.0; }
+			if (length(pixCoord - vec2(0.25, 0.75)) < radius) { distances = distances + 1.0; }
+			if (length(pixCoord - vec2(0.75, 0.75)) < radius) { distances = distances + 1.0; }
+
+			if (distances == 0.0)
 				discard;
-				return;
-			}
-
-			float topLeftDistance = distance(bottomRight, pixCoord);
-
-			if (topLeftDistance > radius - 1.0) {
-				if (primitiveMultisample == 0 && topLeftDistance > radius) {
-					discard;
-					return;
-				} else if (primitiveMultisample == 1) {
-					float distances = 0.0;
-					if (distance(bottomRight, pixCoord + vec2(0.25, 0.25)) < radius) { distances = distances + 1.0; }
-					if (distance(bottomRight, pixCoord + vec2(0.75, 0.25)) < radius) { distances = distances + 1.0; }
-					if (distance(bottomRight, pixCoord + vec2(0.25, 0.75)) < radius) { distances = distances + 1.0; }
-					if (distance(bottomRight, pixCoord + vec2(0.75, 0.75)) < radius) { distances = distances + 1.0; }
-
-					if (distances == 0.0) {
-						discard;
-						return;
-					}
-
-					distances = distances / 4.0;
-
-					)#" + colorVarName + R"#( = )#" + colorVarName + R"#( * distances;
-				}
-			}
+
+			distances = distances / 4.0;
+
+			/* )#" + colorVarName + R"#( = )#" + colorVarName + R"#( * distances; */
```

Comment:
why is this commented out? Where do you set the multisample alpha? I see it working but am confused

Reply:
damn, i forgot that.
I'm trying to fix another thing and i had to find out which shader does what.